src/features/book/book.json-repository.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable prettier/prettier */
+import { CreateBookDto } from './dto/create-book.dto';
+import { UpdateBookDto } from './dto/update-book.dto';
+import defaulteBooks from './book.json'; //массив, работа с массивом
+let books = defaulteBooks;
+import {
+  ConflictException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+@Injectable()
+export class BookJsonRepository {
+  create(createBookDto: CreateBookDto) {
+    const isExsistingBook = books.some((book) => book.id === createBookDto.id);
+    if (isExsistingBook) {
+      throw new ConflictException('same id');
+    }
+    books.push(createBookDto);
+    return books;
+  }
+  findAll() {
+    return books;
+  }
+
+  findOne(id: number) {
+    const condition = (b: CreateBookDto) => b.id === id;
+    const book = books.find(condition);
+    if (!book) {
+      throw new NotFoundException(`book not found!, ${id}`);
+    }
+    return book;
+  }
+  update(id: number, updateBookDto: UpdateBookDto) {
+    books = books.map((book) => {
+      if (book.id === id) {
+        return { ...book, ...updateBookDto };
+      }
+      return this.findOne(id);
+    });
+   // throw new NotFoundException(this.findOne(id));
+  }
+  remove(id: number) {
+    this.findOne(id);
+    const condition = (book) => book.id !== id;
+    const filteredBooks = books.filter(condition);
+    return (books = filteredBooks);
```

Comment:
Please, explain what do you do here

Reply:
1. check for book availability
2. arrow function with a condition for the filter operation
3. call the filter operation for 'books' with a reference to the condition
4. assignment operation

Reply:
Do you think the assignment operation is good as returning a result?

Reply:
yes, it returns the assigned value.

Reply:
Please do not do 2 things at the same time,
1. do assignment operation
2. return value